src/java.base/share/classes/java/time/ZoneOffset.java

```diff
@@ -420,20 +426,40 @@ private static int totalSeconds(int hours, int minutes, int seconds) {
      * @throws DateTimeException if the offset is not in the required range
      */
     public static ZoneOffset ofTotalSeconds(int totalSeconds) {
+        final class Holder {
+            private static final IntFunction<ZoneOffset> ZONE_OFFSET_MAPPER = new ZoneOffsetMapper();
```

Comment:
Can't the ZoneOffsetMapper itself serve as a holder class? so we move this singleton into ZoneOffsetMapper itself.

Reply:
It is possible but this keeps the mapper more local and only accessible where it is supposed to be used. For testing purposed, it might be better to have the class as you propose.

Reply:
If we want it local, I suppose we can convert the whole `ZoneOffsetMapper` local with this singleton in `ofTotalSeconds` static method as well.

Reply:
What I mean is:
```java
final class LazyZoneOffsetMapper implements IntFunction<ZoneOffset> {
    private static final LazyZoneOffsetMapper INSTANCE = new LazyZoneOffsetMapper();

    @Override
    public ZoneOffset apply(int slot) {
        int totalSeconds = slotToSeconds(slot);
        ZoneOffset newValue = new ZoneOffset(totalSeconds);
        ZoneOffset existing = ID_CACHE.putIfAbsent(newValue.getId(), newValue);
        return existing == null ? newValue : existing;
    }
}
```
we can inline this local class so we don't declare two classes nested `ZoneOffsetMapper` and local `Holder` for one purpose. This local class is still lazily loaded only if the `INSTANCE` field is used.